### PR TITLE
Added phpDoc for ApiRequestor, especially regarding thrown errors

### PR DESCRIPTION
**Pull request by rtm-ctrlz**

### CODE REVIEW

File: lib/ApiRequestor.php [hunk 1]
```diff
@@ -9,12 +9,27 @@
...
+     */
     private $_apiKey;
 
+    /**
+     * @var string
+     */
     private $_apiBase;
 
+    /**
+     * @var HttpClient\CurlClient
```
Comment (brandur-stripe) [inline lib/ApiRequestor.php]: I think this is more like a `ClientInterface` rather than necessarily a `CurlClient`.

File: lib/ApiRequestor.php [hunk 2]
```diff
@@ -158,6 +223,13 @@ private static function _specificOAuthError($rbody, $rcode, $rheaders, $resp, $e
         return null;
     }
 
+    /**
+     * @static
+     *
+     * @param $appInfo
```
Comment (brandur-stripe) [inline lib/ApiRequestor.php]: Should we specify a type for this parameter?

**Comment by rtm-ctrlz** [comment]
ptal @brandur-stripe

File: lib/ApiRequestor.php [hunk 3]
```diff
@@ -312,11 +430,19 @@ private function _interpretResponse($rbody, $rcode, $rheaders)
         return $resp;
     }
 
+    /**
+     * @static
+     *
+     * @param HttpClient\CurlClient $client
```
Comment (brandur-stripe) [inline lib/ApiRequestor.php]: I missed this before, but could you also set this one as `HttpClient\ClientInterface`?

File: lib/ApiRequestor.php [hunk 4]
```diff
@@ -312,11 +430,19 @@ private function _interpretResponse($rbody, $rcode, $rheaders)
...
+     */
     public static function setHttpClient($client)
     {
         self::$_httpClient = $client;
     }
 
+    /**
+     * @return HttpClient\CurlClient
```
Comment (brandur-stripe) [inline lib/ApiRequestor.php]: And same here.

**Comment by brandur-stripe** [comment]
Thanks for circling back on this @rtm-ctrlz. I'd missed a couple other instances where `HttpClient\ClientInterface` should be used and added comments on the lines above. Otherwise, LGTM.

**Comment by rtm-ctrlz** [comment]
Fixed CurlClient -> ClientInterface
Also I had to fix phpdoc for ClientInterface interface since @throws and @return sections was broken

ptal @brandur-stripe

**Comment by brandur-stripe** [comment]
> Also I had to fix phpdoc for ClientInterface interface since @throws and @return sections was broken

Oh nice.

LGTM. Thanks!

**Comment by brandur-stripe** [comment]
Released as 6.8.1.

**Pull request merged**